### PR TITLE
1214 map url input/display

### PR DESCRIPTION
**Pull request by Dmac26**
﻿## Summary
Addresses Issue #1214 

This code update adds a new map-url.component and updates all associated .html, .spec.ts, .ts, .js, .es6, and .scss files that allow the creation of a new map url input field. The user input from the field, once submitted in the form of a TOG application is then translated to the database and allowed to be displayed on the application view for both users and admins.

## This pull request is ready to merge when...
- [x] Feature branch starts with the issue number
- [x] Is connected to its original issue via zenhub 👇
- [ ] All tests are passing and meet coverage, linting, and accessibility requirements. And no security vulnerabilities ⚫️(Circle)
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

**Comment by briandavidson** [comment]
Emitting events and adding listeners can be costly and we should avoid using them if we can. We also want to follow the existing patterns in the codebase as much as possible for code consistency.

Can we try removing the blur handler and event emission from the component, and instead leverage the (on change) events that already exist? i.e:
```
this.applicationForm.get('applicantInfo.website').valueChanges.subscribe(website => {
    ...
    do the stuff inside matchUrls()
    ...
});
```

File: frontend/src/app/application-forms/fields/map-url.component.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  @Input() applicantInfo: FormGroup;
+  @Output() blurWebsite: EventEmitter<any> = new EventEmitter();
+
+  constructor(public afs: ApplicationFieldsService) {}
+
+  blurHandler(): void {
+    // function to make blur event bubble up
+    this.blurWebsite.emit();
+  }
+}
```
Comment (briandavidson) [inline frontend/src/app/application-forms/fields/map-url.component.ts:19]: Emitting events and adding listeners can be costly and we should avoid using them if we can. We also want to follow the existing patterns in the codebase as much as possible for code consistency.

Let's remove the blur handler and event emission from the component, and instead leverage the (on change) events that already exist i.e:
```
this.applicationForm.get('applicantInfo.website').valueChanges.subscribe(website => {
    ...
    do the stuff inside matchUrls()
    ...
});
```

File: frontend/src/app/application-forms/temporary-outfitters/temporary-outfitters.component.html [hunk 2]
```diff
@@ -79,15 +79,18 @@ <h3 id="good-standing-evidence-label">{{ goodStandingEvidenceMessage }} <span cl
             [pointOfView]="pointOfView"
             (updateRootDateStatus)="updateDateStatus($event)"></app-activity-description>
 
-          <label id="location-map-label" class="usa-input">Map of proposed activity</label>
-          <p id="location-map-hint-text" class="help-text usa-form-hint">In addition to the description provided above, you may also include a map for further clarification of the specific areas you will be using within the forest.</p>
+          <legend id="location-map-label" class="usa-input location-map-url">Map of proposed activity</legend>
```
Comment (briandavidson) [inline frontend/src/app/application-forms/temporary-outfitters/temporary-outfitters.component.html:82]: What's the reason for switching from `<label>` to `<legend>` here?

Comment (Dmac26) [reply frontend/src/app/application-forms/temporary-outfitters/temporary-outfitters.component.html]: I'm not sure why I did that. I went ahead and reverted to <label>.

File: frontend/src/app/application-forms/fields/map-url.component.ts (hunk 1 above)
Comment (Dmac26) [reply frontend/src/app/application-forms/fields/map-url.component.ts:19]: I've been looking at this, and, honestly I'm not even sure what I was trying to accomplish with this anymore. I removed any use or reference of the blurHandler and noticed no difference in functionality. I must be missing something, but I'm going to move on to your other reviews for now.

**Comment by briandavidson** [comment]
@Dmac26 I'm seeing a few issues on this one:

image

- [ ] `Map of proposed activity` looks like it's bold in the mockup, but not here.
- [ ] The label for the field should not have the word "Test" in it.
- [ ] The label for the field is missing the sub-label text that is in the mockup.
- [ ] `http` URLs are showing as valid, but should be invalid according to the label sub-label text

File: frontend/src/app/application-forms/fields/map-url.component.ts (hunk 1 above)
Comment (briandavidson) [reply frontend/src/app/application-forms/fields/map-url.component.ts]: @Dmac26 once you have made the requested change you can go ahead and hit the `Resolve conversation` button. After you've resolved all the conversations for requested changes you can go ahead and ask for a re-review.

**Comment by Dmac26** [comment]
Closing this PR as it has an incorrect solution on it and the issue has been moved to "Blocked"

**Pull request closed without merging**